tests/testthat/test-tmb-methods.R

```diff
@@ -646,3 +646,149 @@ test_that("response residuals helper function works as expected", {
   expected <- component(object, "y_vector") - as.vector(fitted(object))
   expect_equal(result_rsp, expected)
 })
+
+# simulate.mmrm_tmb ----
+
+test_that("simulate.mmrm returns a data.frame object of correct dimension
+          (including when nsim = 1) and with no missing values", {
+  object <- get_mmrm()
+  set.seed(1001)
+  sims <- simulate(object, nsim = 2)
+  expect_data_frame(sims, any.missing = FALSE, nrows = nrow(object$data), ncols = 2)
+
+  set.seed(202)
+  sims <- simulate(object, nsim = 1)
+  expect_data_frame(sims, any.missing = FALSE, nrows = nrow(object$data), ncols = 1)
+})
+
+test_that("simulate values are correctly centered for standard models", {
+  object <- get_mmrm()
+  set.seed(323)
+  sims <- simulate(object, nsim = 1000)
+  expect_equal(rowMeans(sims), unname(predict(object, object$data)), tolerance = 1e-1)
```

Comment:
can save predict result and then in addition to the means also compare the standard deviation and the interval boundaries (just with 95% confidence level)